cns/restserver/util.go

```diff
@@ -112,9 +113,17 @@ func (service *HTTPRestService) saveNetworkContainerGoalState(req cns.CreateNetw
 	existingNCStatus, ok := service.state.ContainerStatus[req.NetworkContainerid]
 	var hostVersion string
 	var existingSecondaryIPConfigs map[string]cns.SecondaryIPConfig //uuid is key
+	var existingNCVersion string
 	if ok {
 		hostVersion = existingNCStatus.HostVersion
 		existingSecondaryIPConfigs = existingNCStatus.CreateNetworkContainerRequest.SecondaryIPConfigs
+		existingNCVersion = existingNCStatus.VMVersion
```

Comment:
Can we pick it from existingNCStatus.CreateNetworkContainerRequest.Version? Dont know if we set the VMVersion for sure in AKS scenario

Reply:
We dont need existingNCVersion

Reply:
We need existing NC version if we want to update existing IPs from pending programming to available. Only when host version is the same as exsting NC version, CNS can update existing IPs from pending programming to available.
If we want to move pending -> available logic to background thread, then no need of existing NC version.